### ISSUE TITLE
Validation accuracy not correct, but minibatch accuracy during training is fine

### ISSUE THREAD

**Issue by ivanmkc** [tflearn/tflearn#159]
Hi I tried to incorporate tflearn into my existing tensorflow network by replacing 3 hidden layers with tflearn.fully_connected. Following code sample provided here: https://github.com/tflearn/tflearn/blob/master/examples/extending_tensorflow/layers.py

The minibatch accuracy during training is consistent with native TF results, however the validation accuracy is always near zero. Seems like the trained weights are not being used during validation. Any ideas why? 

`

with graph.as_default():
    # Input data.
    tf_train_dataset = tf.placeholder(
        tf.float32, shape=(batch_size, image_size))
    tf_train_labels = tf.placeholder(tf.float32, shape=(batch_size, num_labels))
    tf_valid_dataset = tf.constant(valid_dataset)
    tf_test_dataset = tf.constant(test_dataset)

    # Global step
    global_step = tf.Variable(0, name='global_step', trainable=False)
   
    with tf.name_scope('outputLayer') as scope:
        output_weights = tf.Variable(tf.truncated_normal(
            [num_hidden3, num_labels], stddev=0.1), name='weights')
        output_biases = tf.Variable(tf.constant(1.0, shape=[num_labels]), name='biases')

    # Model.
    def model(data, isUsingDropout):
        net = tflearn.fully_connected(data, num_hidden1, activation='relu')
        net = tflearn.dropout(net, kDropoutProbability)
        net = tflearn.fully_connected(net, num_hidden2, activation='relu')
        net = tflearn.dropout(net, kDropoutProbability)
        net = tflearn.fully_connected(net, num_hidden3, activation='relu')

        return tf.matmul(net, output_weights) + output_biases

    # Training computation.
    logits = model(tf_train_dataset, True)
    loss = tf.reduce_mean(tf.nn.softmax_cross_entropy_with_logits(logits, tf_train_labels), name='xentropy_mean')
    
    # Optimizer.
    optimizer = tf.train.AdamOptimizer().minimize(loss, global_step=global_step)

    # Predictions for the training, validation, and test data.
    train_prediction = tf.nn.softmax(logits)
    valid_prediction = tf.nn.softmax(model(tf_valid_dataset, False))
    test_prediction = tf.nn.softmax(model(tf_test_dataset, False))`

**Comment by aymericdamien** [comment]
It is due to the fact that you are creating 2 distinct models:
```
valid_prediction = tf.nn.softmax(model(tf_valid_dataset, False))
test_prediction = tf.nn.softmax(model(tf_test_dataset, False))
```
This will build your model two times. 
- You can define your model only once, and let 'data' be a placeholder, so that you can feed data multiple times to the same network.
OR
- You can reuse variables by setting scopes:
```
def model(data, isUsingDropout):
    net = tflearn.fully_connected(data, num_hidden1, activation='relu', scope='fc1')
    net = tflearn.dropout(net, kDropoutProbability)
    net = tflearn.fully_connected(net, num_hidden2, activation='relu'. scope='fc2')
    net = tflearn.dropout(net, kDropoutProbability)
    net = tflearn.fully_connected(net, num_hidden3, activation='relu', scope='fc3')
    return tf.matmul(net, output_weights) + output_biases

# And use this function when you need to reuse variables:
tf.get_variable_scope.reuse_variables()
```
